apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -86,77 +89,83 @@ public Coverage transform(MedicareSegment medicareSegment, Beneficiary beneficia
-    Timer.Context timer = createTimerContext("c4dic");
-
-    Coverage coverage = new Coverage();
-    Profile profile = Profile.C4DIC;
+  private void transformC4Dic(Beneficiary beneficiary, Coverage coverage) {
+    addC4DicPayor(coverage);
 
-    addProfile(coverage, profile);
     addC4DicIdentifier(coverage, beneficiary);
-    coverage.setId(CommonTransformerUtils.buildCoverageId(MedicareSegment.C4DIC, beneficiary));
-
     beneficiary.getMedicareBeneficiaryId().ifPresent(coverage::setSubscriberId);
+    coverage.setSubscriber(TransformerUtilsV2.referencePatient(beneficiary));
+    coverage.setStatus(CoverageStatus.ACTIVE);
 
-    setType(coverage);
-    addC4DicPayor(coverage);
-
-    setCoverageRelationship(coverage, SubscriberPolicyRelationship.SELF);
-
-    createCoverageClass(
-        coverage, CoverageClass.GROUP, TransformerConstants.COVERAGE_PLAN, Optional.empty());
-
-    coverage.setBeneficiary(TransformerUtilsV2.referencePatient(beneficiary));
-
-    // update Coverage.meta.lastUpdated
-    TransformerUtilsV2.setLastUpdated(coverage, beneficiary.getLastUpdated());
-
-    timer.stop();
-    return coverage;
+    Extension baseExtension = new Extension();
+    baseExtension.setUrl(TransformerConstants.C4DIC_COLOR_PALETTE_EXT_URL);
+
+    TransformerUtilsV2.addSubExtension(
+        baseExtension,
+        TransformerConstants.C4DIC_FOREGROUNDCOLOR_EXT_URL,
+        new Coding(
+            TransformerConstants.C4DIC_COLORS_CODE_SYSTEM,
+            TransformerConstants.C4DIC_FOREGROUNDCOLOR,
+            null));
+    TransformerUtilsV2.addSubExtension(
+        baseExtension,
+        TransformerConstants.C4DIC_BACKGROUNDCOLOR_EXT_URL,
+        new Coding(
+            TransformerConstants.C4DIC_COLORS_CODE_SYSTEM,
+            TransformerConstants.C4DIC_BACKGROUNDCOLOR,
+            null));
+    TransformerUtilsV2.addSubExtension(
+        baseExtension,
+        TransformerConstants.C4DIC_HIGHLIGHTCOLOR_EXT_URL,
+        new Coding(
+            TransformerConstants.C4DIC_COLORS_CODE_SYSTEM,
+            TransformerConstants.C4DIC_HIGHLIGHTCOLOR,
+            null));
+    coverage.addExtension(baseExtension);
   }
 
   /**
    * Transforms a medicare part A beneficiary into a {@link Coverage} resource.
    *
    * @param beneficiary the {@link Beneficiary} to generate a {@link MedicareSegment#PART_A} {@link
    *     Coverage} resource for
+   * @param enabledProfiles the CARIN {@link Profile} to use
    * @return {@link MedicareSegment#PART_A} {@link Coverage} resource for the specified {@link
    *     Beneficiary}
    */
-  private Coverage transformPartA(Beneficiary beneficiary) {
+  private Coverage transformPartA(Beneficiary beneficiary, Set<Profile> enabledProfiles) {
     Timer.Context timer = createTimerContext("part_a");
     Coverage coverage = new Coverage();
-    Profile profile = Profile.C4BB;
+    Profile profile = (enabledProfiles.contains(Profile.C4DIC)) ? Profile.C4DIC : Profile.C4BB;
```

Comment:
Since we're treating this as an either/or, I think we just need to pass in a single profile rather than a `Set<Profile>`.